--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,7 +10,7 @@
 
 set (booredis_SRCS booredisasync.cpp booredissync.cpp redismessage.cpp)
 
-add_library (booredis STATIC ${booredis_SRCS})
+add_library (booredis SHARED ${booredis_SRCS})
 target_link_libraries(booredis  ${Boost_LIBRARIES})
 
 set_target_properties(booredis PROPERTIES
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,7 @@
 install(FILES
         COPYING
         README
-        DESTINATION share/doc/libbooredis
+        DESTINATION "share/doc/libbooredis${BOOREDIS_LIB_MAJOR_VERSION}"
 )
 
 ADD_CUSTOM_TARGET(uninstall
